### PR TITLE
Allow bind address to be specified (#275)

### PR DESCRIPTION
**Pull request by nareshv**
By default the application listens on loopback address `(127.0.0.1)` and the users can override this behaviour by passing a hostname or ip address to listen on as a last argument to the server.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java [hunk 1]
```diff
@@ -47,6 +48,14 @@ public static void main(String[] args) throws Exception {
         printErrorMessageAndDie();
       }
     }
+    // Future ssl bits
+    String protocol = "http";
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java]: Nit: For clarity, would it be possible to move the hardcoded value to a variable? -- i.e. 
```
private static final String DEFAULT_PROTOCOL = "http";
...
String hostname = DEFAULT_PROTOCOL;
```

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java [hunk 2]
```diff
@@ -47,6 +48,14 @@ public static void main(String[] args) throws Exception {
...
+
+    // For security reasons listen on the loopback address by default
+    String hostname = "127.0.0.1";
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java]: Nit: For clarity, would it be possible to move the hardcoded value to a variable? -- i.e. 
```
private static final String DEFAULT_HOSTNAME = "127.0.0.1";
...
String hostname = DEFAULT_HOSTNAME;
```

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java (hunk 1 above)
Comment (nareshv) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java]: I was following the approach taken for `port` variable. But, lemme fix all the relevant ones as well.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java (hunk 2 above)
Comment (nareshv) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java]: I was following the approach taken for `port` variable. But, lemme fix all the relevant ones as well.

**Pull request merged**